autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/handlers/PopUpFoundMessageHandlerPlugin.kt

```diff
@@ -19,22 +19,44 @@ package com.duckduckgo.autoconsent.impl.handlers
 import android.webkit.WebView
 import com.duckduckgo.autoconsent.api.AutoconsentCallback
 import com.duckduckgo.autoconsent.impl.MessageHandlerPlugin
+import com.duckduckgo.autoconsent.impl.adapters.JSONObjectAdapter
 import com.duckduckgo.autoconsent.store.AutoconsentSettingsRepository
 import com.duckduckgo.di.scopes.AppScope
 import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import timber.log.Timber
 import javax.inject.Inject
 
 @ContributesMultibinding(AppScope::class)
 class PopUpFoundMessageHandlerPlugin @Inject constructor(private val repository: AutoconsentSettingsRepository) : MessageHandlerPlugin {
 
+    private val moshi = Moshi.Builder().add(JSONObjectAdapter()).build()
+
     override fun process(messageType: String, jsonString: String, webView: WebView, autoconsentCallback: AutoconsentCallback) {
-        if (supportedTypes.contains(messageType)) {
-            if (repository.userSetting) {
-                return
+        try {
+            if (supportedTypes.contains(messageType)) {
+                val message: PopUpFoundMessage = parseMessage(jsonString) ?: return
+                if (repository.userSetting) return
+                if (message.cmp.endsWith(IGNORE_CMP_SUFFIX)) return
```

Comment:
Just flagging to be absolutely sure that case is important here and we don't need to ignore it

Reply:
Good catch! Fixed